third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -2252,6 +2252,22 @@ impl Frame {
                                 vec![],
                             )
                             .map(Rc::new)?;
+
+                        let captured_tys = mask.extract(function.param_tys(), true);
+                        for ty in captured_tys {
+                            interpreter.ty_depth_checker.check_depth_of_type(
```

Comment:
Note: depth checker is currently not correct wrt to function values because it checks depth of arg/return types (same problem as layouts that I fixed before), that's probably why some of the tests fail. Why checking depth here for these (this is not related to gas?)

Also, we check the type of already constructed values, so it will always pass (assuming bug above is fixed)?  We need to check depth + 1 probably, but I am experimenting with removing check here and checking values depths instead.